compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3700,7 +3700,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             // See if the `Type::function(val)` where `function` wasn't found corresponds to a
             // `Trait` that is imported directly, but `Type` came from a different version of the
             // same crate.
-            let rcvr_ty = self.tcx.type_of(def_id).instantiate_identity();
+
+            let rcvr_ty = match self.tcx.def_kind(def_id) {
```

Comment:
Leave a comment explaining what kinds of `DefKind`s you're actually trying to avoid here. It makes it look less like you just forgot some special cases.

Reply:
Also, why *are* we doing this? Why can't we just look at the `node_ty` of the `TyKind::Path` above? That would handle the non-ADT cases too, no?

Reply:
done